src/utils/inferencedata.jl

```diff
@@ -1,17 +1,83 @@
-using InferenceObjects
-using PosteriorDB
+function inferencedata(m::SampleModel; kwargs...)
 
-import Base: convert
+    # Used keyword arguments
+    used_keywords = [:include_log_likelihood, :include_sample_stats,
+        :include_posterior_predictive]
 
-"""
+    # Default is only posterior values
+    include_log_likelihood = false
+    include_sample_stats = false
+    include_posterior_predictive = false
 
-# inference
+    # Warmup included if stored
+    include_warmup = m.save_warmup
 
-# Convert the output file(s) created by cmdstan to a InferenceData object.
+    for (indx, kwd) in enumerate(kwargs)
+        if kwd[1] == :include_log_likelihood
+            include_log_likelihood = kwargs[indx][1] ? true : false
+        end
+        if kwd[1] == :include_sample_stats
+            include_sample_stats = kwargs[indx][1] ? true : false
+        end
+        if kwd[1] == :include_posterior_predictive
+            include_posterior_predictive = kwargs[indx][1] ? true : false
+        end
+    end
 
-$(SIGNATURES)
+    stan_nts = read_samples(m, :namedtuples; include_internals=include_sample_stats)
```

Comment:
I think it makes sense to apply the `stan_key_map` here before calling `from_namedtuple`:
```suggestion
    stan_nts = read_samples(m, :namedtuples; include_internals=include_sample_stats)
    stan_key_map = (
            n_leapfrog__=:n_steps,
            treedepth__=:tree_depth,
            energy__=:energy,
            lp__=:lp,
            stepsize__=:step_size,
            divergent__=:diverging,
            accept_stat__=:acceptance_rate,
        );
    stan_sample_nts = NamedTuple{filter(∉(keys(stan_key_map)), keys(stan_nts))}(stan_nts)
    stan_stats_nts = NamedTuple{filter(∈(keys(stan_key_map)), keys(stan_nts))}(stan_nts)
    stan_stats_nts_rekey = NamedTuple{map(Base.Fix1(getproperty, stan_key_map), keys(stan_stats_nts))}(values(stan_stats_nts))
    idata = from_namedtuple(stan_sample_nts; sample_stats=stan_stats_nts_rekey, kwargs...)
```
Code could be made a lot simpler with some helper functions.